File: src/main/java/org/hwabeag/divesystem/inventorys/DiveSystemItemSettingGUI.java
package org.hwabeag.divesystem.inventorys;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.hwabeag.divesystem.config.ConfigManager;
import org.jetbrains.annotations.Nullable;

import java.util.Arrays;
import java.util.Objects;

public class DiveSystemItemSettingGUI implements Listener {
    private final Inventory inv;

    FileConfiguration DiveSystemConfig = ConfigManager.getConfig("dive-system");
    FileConfiguration PlayerConfig = ConfigManager.getConfig("player");

    private void initItemSetting(Player p) {
        String name = p.getName();
        int PlayerPage = PlayerConfig.getInt(name + ".페이지");
        String PlayerShop = PlayerConfig.getString(name + ".설정상점");
        int index = DiveSystemConfig.getInt("잠수상점." + PlayerShop + ".라인");
        int N = 0;
        int Page = 0;
        if (DiveSystemConfig.getConfigurationSection("잠수상점." + PlayerShop + ".물품." + PlayerPage) != null) {
            for (String key : Objects.requireNonNull(DiveSystemConfig.getConfigurationSection("잠수상점." + PlayerShop + ".물품." + PlayerPage)).getKeys(false)) {
                if (Page == PlayerPage) {
                    @Nullable ItemStack item = DiveSystemConfig.getItemStack("잠수상점." + PlayerShop + ".물품." + PlayerPage + "." + key);
                    inv.setItem(Integer.parseInt(key), item);
                }
                N = N + 1;
                if (N >= index * 9) {
                    Page = Page + 1;
                    N = 0;
                }
            }
        }
        ItemStack item = new ItemStack(Material.PAPER, 1);
        ItemMeta itemMeta = item.getItemMeta();
        itemMeta.setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a이전 페이지"));
        itemMeta.setLore(Arrays.asList(ChatColor.translateAlternateColorCodes('&', "&a- &f클릭 시 이전 페이지로 이동합니다.")));
        item.setItemMeta(itemMeta);
        inv.setItem(index*9+9-3,item);

        item = new ItemStack(Material.PAPER, 1);
        itemMeta = item.getItemMeta();
        itemMeta.setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a다음 페이지"));
        itemMeta.setLore(Arrays.asList(ChatColor.translateAlternateColorCodes('&', "&a- &f클릭 시 다음 페이지로 이동합니다.")));
        item.setItemMeta(itemMeta);
        inv.setItem(index*9+9-1,item);
    }

    public DiveSystemItemSettingGUI(Player p) {
        String name = p.getName();
        String PlayerShop = PlayerConfig.getString(name + ".설정상점");
        int index = DiveSystemConfig.getInt("잠수상점." + PlayerShop + ".라인");
        inv = Bukkit.createInventory(null,index*9+9,"물품설정");
        this.initItemSetting(p);
    }

    public void open(Player player){
        player.openInventory(inv);
    }

}

File: src/main/kotlin/org/hwabeag/divesystem/inventorys/DiveSystemItemSettingGUI.kt
package org.hwabeag.divesystem.inventorys

import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.entity.Player
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack
import org.hwabeag.divesystem.config.ConfigManager

class DiveSystemItemSettingGUI(player: Player) {
    private val diveSystemConfig: FileConfiguration = ConfigManager.getConfig("dive-system")
    private val playerConfig: FileConfiguration = ConfigManager.getConfig("player")
    private val inv: Inventory

    init {
        val name = player.name
        val playerShop = playerConfig.getString("$name.설정상점") ?: "없음"
        val line = diveSystemConfig.getInt("잠수상점.$playerShop.라인")
        inv = Bukkit.createInventory(null, line * 9 + 9, "물품설정")
        initItemSetting(player)
    }

    private fun initItemSetting(player: Player) {
        val name = player.name
        val page = playerConfig.getInt("$name.페이지")
        val playerShop = playerConfig.getString("$name.설정상점") ?: return
        val line = diveSystemConfig.getInt("잠수상점.$playerShop.라인")
        val section = diveSystemConfig.getConfigurationSection("잠수상점.$playerShop.물품.$page")
        section?.getKeys(false)?.forEach { key ->
            val item = diveSystemConfig.getItemStack("잠수상점.$playerShop.물품.$page.$key")
            inv.setItem(key.toInt(), item)
        }

        val prev = ItemStack(Material.PAPER, 1)
        prev.itemMeta = prev.itemMeta?.apply {
            setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a이전 페이지"))
            lore = listOf(ChatColor.translateAlternateColorCodes('&', "&a- &f클릭 시 이전 페이지로 이동합니다."))
        }
        inv.setItem(line * 9 + 6, prev)

        val next = ItemStack(Material.PAPER, 1)
        next.itemMeta = next.itemMeta?.apply {
            setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a다음 페이지"))
            lore = listOf(ChatColor.translateAlternateColorCodes('&', "&a- &f클릭 시 다음 페이지로 이동합니다."))
        }
        inv.setItem(line * 9 + 8, next)
    }

    fun open(player: Player) {
        player.openInventory(inv)
    }
}
